### PR TITLE
fix: set the size correctly for real part in slicing function

### PR DESCRIPTION
**Pull request by hamed-musallam**
We have encountered a major bug that causes failures in the slicing tool and 2D phase correction tool. This issue was introduced by the PR 
- https://github.com/cheminfo/nmrium/pull/2949

### CODE REVIEW

File: src/data/data2d/Spectrum2D/getSlice.ts [hunk 1]
```diff
@@ -104,7 +104,7 @@ export function getSlice(
 function initiateData(from: number, to: number, size: number): NmrData1D {
   return {
     x: xSequentialFillFromTo({ from, to, size }),
-    re: new Float64Array(length),
+    re: new Float64Array(size),
```
Comment (targos) [inline src/data/data2d/Spectrum2D/getSlice.ts:107]: How was `length` not reported as non-existing variable?

Comment (hamed-musallam) [reply src/data/data2d/Spectrum2D/getSlice.ts:107]: it is strange !!!!  the `"strict": true` is in the tsconfig` but it is not reported

Comment (targos) [reply src/data/data2d/Spectrum2D/getSlice.ts:107]: Found it: it's one of the many available globals on `window`:
image

I'll add it to our [list of restricted globals](https://github.com/cheminfo/eslint-config-cheminfo-react/blob/main/base.js#L24-L31)

Comment (hamed-musallam) [reply src/data/data2d/Spectrum2D/getSlice.ts:107]: It is possible that length is defined as a global variable somewhere, but I've tried searching for it and couldn't find where it's defined

**Comment by hamed-musallam** [comment]
@lpatiny

After merging this PR, we'll need to create a new release of NMRium

File: src/data/data2d/Spectrum2D/getSlice.ts (hunk 1 above)
Comment (hamed-musallam) [reply src/data/data2d/Spectrum2D/getSlice.ts:107]: Alright, I'll mark it as resolved and merge this PR. Updating ESLint rules can be done in another PR. What do you think?

Comment (targos) [reply src/data/data2d/Spectrum2D/getSlice.ts:107]: Yes, of course

**Pull request merged**